Source/Core/VideoCommon/TextureDecoder_x64.cpp

```diff
@@ -28,32 +28,33 @@ static inline u32 DecodePixel_IA8(u16 val)
 
 static inline u32 DecodePixel_RGB565(u16 val)
 {
-  int r, g, b, a;
-  r = Convert5To8((val >> 11) & 0x1f);
-  g = Convert6To8((val >> 5) & 0x3f);
-  b = Convert5To8((val)&0x1f);
-  a = 0xFF;
-  return r | (g << 8) | (b << 16) | (a << 24);
+  int r, g, b;
+
+  r = val & 0xf8;
```

Comment:
Is the difference in performance worth the reduced clarity here? It might show up when just calling the texture decoding functions in a loop, but overall texture decoding is quite low in overhead.